tools/bazar/controllers/EntryController.php

```diff
@@ -254,10 +254,8 @@ private function getValuesForCustomTemplate($entry, $form)
             }
         }
         
-        try {
-            $html['semantic'] = $GLOBALS['wiki']->services->get(SemanticTransformer::class)->convertToSemanticData($form['bn_id_nature'], $html, true);
-        } catch (\Exception $e) {
-            // Do nothing if semantic type is not available
+        if ($form['bn_sem_type']){
+            $html['semantic'] = $GLOBALS['wiki']->services->get(SemanticTransformer::class)->convertToSemanticData($form, $html, true);
         }
```

Comment:
je me permet de revenir  cette PR pour ne pas qu'elle tombe dans l'oubli.

@srosset81 c'est en voyant ce catch que je m'étais motivé à bonifier le code (y'a du boulot dans yeswiki, donc ça me paraît important qu'on le fasse par itération ;))
pour moi, c'est une mauvaise pratique d'utiliser la récupération d'une Exception pour traiter un cas classique. Parfois, on utilise des fonctions d'une librairie donc on n'a pas le choix, mais autant éviter qd c'est possible. Aussi l'Exception n'est pas spécifique et cela pourrait rendre silencieux d'autres exceptions qui se produisent.